clang/test/CodeGen/aie/aie2/aie2-scl2vec-intrinsic.cpp

```diff
@@ -556,9 +556,7 @@ v16float test_broadcast_float(float b) {
 
 // CHECK-LABEL: @_Z16test_shiftl_elemDv64_ai(
 // CHECK-NEXT:  entry:
-// CHECK-NEXT:    [[SEXT:%.*]] = shl i32 [[S:%.*]], 24
-// CHECK-NEXT:    [[CONV_I1:%.*]] = ashr exact i32 [[SEXT]], 24
-// CHECK-NEXT:    [[TMP0:%.*]] = tail call noundef <64 x i8> @llvm.aie2.vbroadcast8.I512(i32 [[CONV_I1]])
+// CHECK-NEXT:    [[TMP0:%.*]] = tail call noundef <64 x i8> @llvm.aie2.vbroadcast8.I512(i32 [[S:%.*]])
 // CHECK-NEXT:    [[TMP1:%.*]] = bitcast <64 x i8> [[V:%.*]] to <16 x i32>
```

Comment:
@gbossu Without DemandedBits these wont be eliminated

Reply:
Let's see with @konstantinschwarz if he has some concerns with `zext`, to me the code looks correct.

Reply:
I think its fine. Having cleaner LLVM IR potentially allows other optimizations to kick in.
The additional instruction selection patterns don't hurt and cover the cases where some extensions still survive the IR optimizers.